src/storage/v2/edge_accessor.cpp

```diff
@@ -184,15 +186,18 @@ Result<std::vector<std::tuple<PropertyId, PropertyValue, PropertyValue>>> EdgeAc
 
   if (edge_.ptr->deleted) return Error::DELETED_OBJECT;
 
+  bool should_apply_delta_if_identical_update = storage_->config_.salient.items.delta_on_identical_property_update;
   using ReturnType = decltype(edge_.ptr->properties.UpdateProperties(properties));
   std::optional<ReturnType> id_old_new_change;
-  utils::AtomicMemoryBlock atomic_memory_block{
-      [transaction_ = transaction_, edge_ = edge_, &properties, &id_old_new_change]() {
-        id_old_new_change.emplace(edge_.ptr->properties.UpdateProperties(properties));
-        for (auto &[property, old_value, new_value] : *id_old_new_change) {
-          CreateAndLinkDelta(transaction_, edge_.ptr, Delta::SetPropertyTag(), property, std::move(old_value));
-        }
-      }};
+  utils::AtomicMemoryBlock atomic_memory_block{[transaction_ = transaction_, edge_ = edge_, &properties,
+                                                &id_old_new_change, should_apply_delta_if_identical_update]() {
+    id_old_new_change.emplace(edge_.ptr->properties.UpdateProperties(properties));
+    for (auto &[property, old_value, new_value] : *id_old_new_change) {
+      if (old_value != new_value || should_apply_delta_if_identical_update) {
+        CreateAndLinkDelta(transaction_, edge_.ptr, Delta::SetPropertyTag(), property, std::move(old_value));
+      }
```

Comment:
Avoid `||`, `!`, and `!=`.
Prefer `&&`, `==`
```suggestion
      if (skip_duplicate_write && old_value == new_value) continue;
      CreateAndLinkDelta(transaction_, edge_.ptr, Delta::SetPropertyTag(), property, std::move(old_value));
      
```